src/hooks/job-queues/use-submit-job-queue.js

```diff
@@ -0,0 +1,36 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import history from '../../services/history'
+import formatError from '../../services/format-error'
+
+const createMutation = (name) => ({
+    resource: `scheduler/queues/${name}`,
+    type: 'create',
+    data: ({ queue }) => queue,
+})
+
+const useSubmitJobQueue = () => {
+    const engine = useDataEngine()
+    const submitJobQueue = (queue) => {
+        const mutation = createMutation(queue.name)
+        return engine
+            .mutate(mutation, { variables: { queue } })
+            .then(() => {
+                history.push('/')
+            })
+            .catch((error) => {
+                const isValidationError = error.type === 'access'
+
+                // Potential validation error, return it in a format final-form can handle
+                if (isValidationError) {
+                    return formatError(error)
+                }
+
+                // Throw any unexpected errors
+                throw error
```

Comment:
do you want to throw this error? It seems the error needs need to be returned to be be picked up by the react final form?

Reply:
Yeah this is on purpose. The idea is based on this section of the FF docs:

![image](https://github.com/dhis2/scheduler-app/assets/7355199/ef4363ea-6894-44e2-bfe7-7e4fa68210d0)
> https://final-form.org/docs/final-form/types/Config#onsubmit

So anything that is not a server-side validation error will reject the promise, as they're suggesting here. In looking at this though, I don't think that this promise rejection is dealt with anywhere. We should make sure that we display this in a noticebox. Good catch, I thought we were already handling this, but it doesn't look like it.

Reply:
Created an issue for this: https://dhis2.atlassian.net/browse/DHIS2-15556. @kabaros, what do you think? Shall we handle this across the entire app before merging?

Reply:
I suppose for simplicity we could just handle these errors with final-forms FORM_ERROR key. Final form does not supply a convenient mechanism for handling the errors from onSubmit promise rejections. So handling them the same way as generic form errors seems appropriate in the absence of anything else.

Reply:
I've gone ahead and went with the last approach. Should be resolved with the last 3 commits.